Realm/Realm/Realm.cs

```diff
@@ -748,11 +748,11 @@ public T Write<T>(Func<T> function)
         /// Action to execute inside a <see cref="Transaction"/>, creating, updating, or removing objects.
         /// </param>
         /// <returns>An awaitable <see cref="Task"/>.</returns>
-        public Task WriteAsync(Action<Realm> action)
+        public async Task WriteAsync(Action<Realm> action)
         {
             Argument.NotNull(action, nameof(action));
 
-            return WriteAsync(tempRealm =>
+            await WriteAsync(tempRealm =>
```

Comment:
I don't think we need to change this. Not that it's that big of a deal but there's no need to allocate a state machine for this method.